equinix/fabric_port_read_schema.go

```diff
@@ -0,0 +1,350 @@
+			Computed:    true,
+			Description: "Link aggregation group name",
+		},
+		"member_status": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "LAG port status",
+		},
+	}
+}
+
+var readPortSettingsRes = &schema.Resource{
+	Schema: readFabricPortSettings(),
+}
+
+func readFabricPortSettings() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"port_type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port type",
+		},
+	}
+}
+
+func readPortOperationSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"operational_status": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port operation status",
+		},
+		"connection_count": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Total number of current connections",
+		},
+		"op_status_changed_at": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Date and time at which port availability changed",
+		},
+	}
+}
+
+var readPortsRedundancyRes = &schema.Resource{
+	Schema: readPortsRedundancySch(),
+}
+
+func readPortsRedundancySch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"enabled": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Access point redundancy",
+		},
+		"group": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Port redundancy group",
+		},
+		"priority": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Priority type-Primary or Secondary",
+		},
+	}
+}
+
+var readPortTetherRes = &schema.Resource{
+	Schema: readFabricPortTether(),
+}
+
+func readFabricPortTether() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"cross_connect_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port cross connect identifier",
+		},
+		"cabinet_number": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port cabinet number",
+		},
+		"system_name": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port system name",
+		},
+		"patch_panel": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port patch panel",
+		},
+		"patch_panel_port_a": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port patch panel port A",
+		},
+		"patch_panel_port_b": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port patch panel port B",
+		},
+	}
+}
+
+func readFabricPortResourceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port type",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port URI information",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
```

Comment:
"uuid" must be `Required` for `equinix_fabric_port` data source. However, since it is optional for  `equinix_fabric_ports` I would add a separate function to retrieve/modify `equinix_fabric_ports` schema

```
func readFabricPortsResourceSchema() map[string]*schema.Schema {
	sch := readFabricPortResourceSchema()
	sch["uuid"].Optional = true
	sch["uuid"].Required = false
	return sch
}
```

And reference it in

https://github.com/equinix/terraform-provider-equinix/pull/230/files#diff-f9ecc09f3eb05f0a73dfdafa8f04b48297432eb40fab00a66b7b070daa5a646aR338